fe2-android/app/src/main/java/com/github/dedis/student20_pop/qrcode/QRCodeScanningFragment.java

```diff
@@ -146,13 +138,37 @@ private void startCamera() throws SecurityException {
       }
     }
   }
-  private void setupCloseRollCallButton() {
+
+  private void setupCloseRollCallButtons() {
     mQrCodeFragBinding.addAttendeeConfirm.setOnClickListener(
-            v -> ((LaoDetailViewModel)mQRCodeScanningViewModel).closeRollCall());
+            clicked -> setupClickCloseListener(R.id.fragment_lao_detail)
+    );
+    //observe events that require current open rollcall to be closed
+    ((LaoDetailViewModel)mQRCodeScanningViewModel)
+            .getAskCloseRollCallEvent()
+            .observe(
+                    this,
+                    integerEvent -> {
+                      Integer nextFragment = integerEvent.getContentIfNotHandled();
+                      if (nextFragment != null) {
+                        setupClickCloseListener(nextFragment);
+                      }
+                    });
```

Comment:
This doesn't really correspond to the setup of the close roll call button, maybe you can create another method for it and call it on `onCreateView`